elastic/beats
# Add linux/rapl metricset

## What does this PR do?

This PR adds a `linux/rapl` metricset that reports CPU power data for intel CPUs via the RAPL interface: https://01.org/blogs/2014/running-average-power-limit-%E2%80%93-rapl

This is a fairly basic metricset, and we may want to expand it with:

- Dashboards
- A gauge that reports raw usage in Joules

## Why is it important?

This is a somewhat niche, but still fun feature that reports power usage.

## Checklist


## How to test this PR locally

- Download and build on a newer Intel system that supports RAPL (sandy bridge or later)
- Run the metricset, make sure it works.
